tests/phpunit/CiviTest/CiviUnitTestCase.php

```diff
@@ -2775,8 +2775,24 @@ public function paymentProcessorCreate($params = array()) {
 
   /**
    * Set up initial recurring payment allowing subsequent IPN payments.
+   *
+   * @param array $recurParams (Optional)
+   * @param array $contributionParams (Optional)
    */
-  public function setupRecurringPaymentProcessorTransaction($params = array()) {
+  public function setupRecurringPaymentProcessorTransaction($recurParams = [], $contributionParams = []) {
+    $contributionParams = array_merge([
+        'total_amount' => '200',
+        'invoice_id' => $this->_invoiceID,
+        'financial_type_id' => 1,
```

Comment:
can you change this to 'financial_type_id' => 'Donation',

Reply:
Wouldn't it be more appropriate to use something like CRM_Core_Pseduoconstant::value(‘CRM_Contribute_BAO_Contribution’, ‘financial_type_id’, 'Donation’)

Reply:
Done. @JoeMurray Normally yes. In this case we can use API "magic" so can set 'Donation' directly and the API will work it out.